### PR TITLE
Badge: exclude duplicated builds

### PR DESCRIPTION
**Pull request by stsewd**
ref https://github.com/readthedocs/readthedocs.org/issues/8015

### CODE REVIEW

File: readthedocs/projects/views/public.py [hunk 1]
```diff
@@ -177,10 +177,13 @@ def get(self, request, project_slug, *args, **kwargs):
             ).first()
 
         if version:
-            last_build = version.builds.filter(
-                type='html',
-                state='finished',
-            ).order_by('-date').first()
+            last_build = (
+                version.builds
+                .filter(type='html', state='finished')
```
Comment (ericholscher) [inline readthedocs/projects/views/public.py:182]: I feel like we have another bug where builds are default set to `finished` when they are created. We should probably change that as well so we don't hit this as well 👍

**Review by ericholscher — APPROVED** [review]
Seems reasonable 👍

File: readthedocs/projects/views/public.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/projects/views/public.py:182]: Do you mean this is another problem (not something I need to change here)?

Comment (ericholscher) [reply readthedocs/projects/views/public.py:182]: Yea, it can be done in another PR.

**Pull request merged**